src/wasm/WasmModule.cpp

```diff
@@ -260,6 +276,70 @@ uint32_t WasmModule::mapSharedStateMemory(
     return sharedMemWasmPtrs[segmentKey];
 }
 
+uint32_t WasmModule::getCurrentBrk()
+{
+    return currentBrk;
+}
+
+uint32_t WasmModule::createMemoryGuardRegion()
+{
+    auto logger = faabric::util::getLogger();
+
+    size_t regionSize = GUARD_REGION_SIZE;
+    uint32_t wasmOffset = growMemory(regionSize);
+
+    uint8_t* nativePtr = wasmPointerToNative(wasmOffset);
+
+    // NOTE: we want to protect these regions from _writes_, but we don't want
+    // to stop them being read, otherwise snapshotting will fail. Therefore we
+    // make them read-only
+    int res = mprotect(nativePtr, regionSize, PROT_READ);
+    if (res != 0) {
+        logger->error("Failed to create memory guard: {}",
+                      std::strerror(errno));
+        throw std::runtime_error("Failed to create memory guard");
+    }
+
+    logger->debug(
+      "Created guard region {}-{}", wasmOffset, wasmOffset + regionSize);
+
+    return wasmOffset;
+}
+
+void WasmModule::createThreadStackPool()
```

Comment:
This sets up our pool of thread stacks. It's called early on so they live at the bottom of the heap in a bunch.